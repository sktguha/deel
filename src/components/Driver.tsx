import { useState } from "react";
import { AutoComplete } from "./AutoComplete";

export function Driver({single ,required, errorTextProp}) {
  const defaultOptions = [{
    label: "Hello",
    id: 1
  },
  {
    label: "Help",
    id: 2
  },
  {
    label: "Helldivers",
    id: 3
  }
  ];
  const [selectedOptions, setSelectedOptions] = useState([]);
  const [errorText, setErrorText] = useState(required ? "This is a required field" : false);
  return <AutoComplete errorText={errorTextProp || errorText} single={single} selectedOptions={selectedOptions}
    getOptions={(text) => new Promise((res) => setTimeout(() => res(defaultOptions), Math.random() * 2500))}
    onSelectedOptionsChange={(options) => {
      console.log("new options is ", options);
      if (required && options.length === 0) { 
        setErrorText("This is a required field");
      } else {
        setErrorText(false);
      }
      setSelectedOptions(options);
    }} />;
}
