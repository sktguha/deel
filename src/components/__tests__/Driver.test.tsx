import React from 'react';
import { Driver } from "../Driver"
import { render , screen } from '@testing-library/react';


test("Driver base",()=>{
    const { container } = render(<Driver />)
    expect(container).toMatchSnapshot();
})

test("Driver single prop",()=>{
  const { container } = render(<Driver single/>)
  expect(container).toMatchSnapshot();
})

test("Driver required",()=>{
  const { container } = render(<Driver required/>)
  expect(container).toMatchSnapshot();
})

test("Driver errorText",()=>{
  const { container } = render(<Driver errorTextProp={"custom errorText for test"}/>)
  expect(container).toMatchSnapshot();
})