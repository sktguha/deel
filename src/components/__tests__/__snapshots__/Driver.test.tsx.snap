// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Driver base 1`] = `
<div>
  <div
    class="Autocomplete"
  >
    <div>
      Autocomplete
    </div>
    <div
      class="input-container"
    >
      <input
        class="input-box false"
        placeholder="autocomplete demo"
        tabindex="-1"
        type="text"
      />
    </div>
    <div
      class="selected-options"
    />
    <div>
      <div
        class="available-options-to-select"
      />
    </div>
  </div>
</div>
`;

exports[`Driver errorText 1`] = `
<div>
  <div
    class="Autocomplete"
  >
    <div>
      Autocomplete
    </div>
    <div
      class="error-text"
    >
      custom errorText for test
    </div>
    <div
      class="input-container"
    >
      <input
        class="input-box input-box-error"
        placeholder="autocomplete demo"
        tabindex="-1"
        type="text"
      />
    </div>
    <div
      class="selected-options"
    />
    <div>
      <div
        class="available-options-to-select"
      />
    </div>
  </div>
</div>
`;

exports[`Driver required 1`] = `
<div>
  <div
    class="Autocomplete"
  >
    <div>
      Autocomplete
    </div>
    <div
      class="error-text"
    >
      This is a required field
    </div>
    <div
      class="input-container"
    >
      <input
        class="input-box input-box-error"
        placeholder="autocomplete demo"
        tabindex="-1"
        type="text"
      />
    </div>
    <div
      class="selected-options"
    />
    <div>
      <div
        class="available-options-to-select"
      />
    </div>
  </div>
</div>
`;

exports[`Driver single prop 1`] = `
<div>
  <div
    class="Autocomplete"
  >
    <div>
      Autocomplete
    </div>
    <div
      class="input-container"
    >
      <input
        class="input-box false"
        placeholder="autocomplete demo"
        tabindex="-1"
        type="text"
      />
    </div>
    <div
      class="selected-options"
    />
    <div>
      <div
        class="available-options-to-select"
      />
    </div>
  </div>
</div>
`;
